packages/roosterjs-content-model-api/lib/modelApi/list/setListType.ts

```diff
@@ -156,3 +165,37 @@ function shouldIgnoreBlock(block: ReadonlyContentModelBlock) {
             return true;
     }
 }
+
+function adjustIndentation(listItem: ShallowMutableContentModelListItem) {
+    const block = listItem.blocks[0];
+    if (
+        block.blockType == 'Paragraph' &&
+        block.segments.length > 0 &&
+        block.segments[0].segmentType == 'Text'
+    ) {
+        const spaces = countSpacesBeforeText(block.segments[0].text);
+        const tabSpaces = Math.floor(spaces / 4);
+        if (tabSpaces > 0) {
+            mutateSegment(block, block.segments[0], textSegment => {
+                textSegment.text = textSegment.text.substring(tabSpaces * 4);
+            });
+            if (tabSpaces) {
+                listItem.levels[0].format.marginLeft = tabSpaces * IndentStepInPixel + 'px';
+            }
+        }
+    }
+}
+
+function countSpacesBeforeText(str: string) {
+    let count = 0;
+
+    for (const char of str) {
+        if (char === SPACE) {
```

Comment:
what about `&nbsp;`?

Reply:
We count the spaces of the text of text segment. Do we need to check `&nbsp`; when not handling HTML directly?

Reply:
I mean, space (\u20) and `&nbsp;` (\ua0) are different characters. We need to handle both of them.

Reply:
Actually, the space we added is `&ensp;`, or say \u2002. So that your code can work since you are also checking \u2002.

My question is, do we want to support the spaces added by user manually, in that case it could be \u20 or \ua0. If not, then current code change is fine. Otherwise, we need to check all of them.

Reply:
I got it now, I'll reach Jorge and ask this question.